packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -723,40 +733,61 @@ class _RangeSliderState extends State<RangeSlider> with TickerProviderStateMixin
       mouseCursor: effectiveMouseCursor,
       child: CompositedTransformTarget(
         link: _layerLink,
-        child: _RangeSliderRenderObjectWidget(
-          values: _unlerpRangeValues(widget.values),
-          divisions: widget.divisions,
-          labels: widget.labels,
-          sliderTheme: sliderTheme,
-          textScaleFactor: effectiveTextScale,
-          screenSize: screenSize(),
-          onChanged: _enabled && (widget.max > widget.min) ? _handleChanged : null,
-          onChangeStart: widget.onChangeStart != null ? _handleDragStart : null,
-          onChangeEnd: widget.onChangeEnd != null ? _handleDragEnd : null,
-          state: this,
-          semanticFormatterCallback: widget.semanticFormatterCallback,
-          hovering: _hovering,
+        child: OverlayPortal(
+          controller: _valueIndicatorOverlayPortalController,
+          overlayChildBuilder: (BuildContext context) {
+            if (sliderTheme.showValueIndicator == ShowValueIndicator.alwaysVisible) {
```

Comment:
This should be a switch statement. Might be worth moving out of the widget tree to a method for better readability.